tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -453,11 +459,38 @@ export class WebGPUBackend extends KernelBackend {
 
     const encoder = this.device.createCommandEncoder();
     const pass = encoder.beginComputePass();
+    if (shouldTimeProgram) {
+      if (env().getBool('WEBGPU_QUERY_API_ENABLED')) {
+        querySet = this.device.createQuerySet({
+          type: "timestamp",
+          count: 2,
+        });
+        queryBuffer = this.device.createBuffer({
+          size: 16,
+          usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.QUERY_RESOLVE,
+        });
+        pass.writeTimestamp(querySet, 0);
+      } else {
+        query = this.startTimer();
+      }
+    }
     pass.setPipeline(pipeline);
     pass.setBindGroup(0, bg);
     pass.dispatch(
         program.dispatch[0], program.dispatch[1], program.dispatch[2]);
-    pass.endPass();
+    if (shouldTimeProgram) {
+      if (env().getBool('WEBGPU_QUERY_API_ENABLED')) {
+        pass.writeTimestamp(querySet, 1);
+        pass.endPass();
```

Comment:
Move `pass.endPass();` out of `if (shouldTimeProgram)` so that you don't need to call it in each branch.

Reply:
Good idea, solved this problem and I moved resolveQuerySet to function named getTimeFromQuerySet().